pkg/apis/options/providers.go

```diff
@@ -0,0 +1,177 @@
+package options
+
+import "github.com/oauth2-proxy/oauth2-proxy/v7/providers"
+
+// Providers is a collection of definitions for providers.
+type Providers []Provider
+
+// Provider holds all configuration for a single provider
+type Provider struct {
+	// ClientID is the OAuth Client ID that is defined in the provider
+	// This value is required for all providers.
+	ClientID string `json:"clientID,omitempty"`
+	// ClientSecret is the OAuth Client Secret that is defined in the provider
+	// This value is required for all providers.
+	ClientSecret string `json:"clientSecret,omitempty"`
+	// ClientSecretFile is the name of the file
+	// containing the OAuth Client Secret, it will be used if ClientSecret is not set.
+	ClientSecretFile string `json:"clientSecretFile,omitempty"`
+
+	// KeycloakConfig holds all configurations for Keycloak provider.
+	KeycloakConfig KeycloakOptions `json:"keycloakConfig,omitempty"`
+	// AzureConfig holds all configurations for Azure provider.
+	AzureConfig AzureOptions `json:"azureConfig,omitempty"`
+	// BitbucketConfig holds all configurations for Bitbucket provider.
+	BitbucketConfig BitbucketOptions `json:"bitbucketConfig,omitempty"`
+	// GitHubConfig holds all configurations for GitHubC provider.
+	GitHubConfig GitHubOptions `json:"githubConfig,omitempty"`
+	// GitLabConfig holds all configurations for GitLab provider.
+	GitLabConfig GitLabOptions `json:"gitlabConfig,omitempty"`
+	// GoogleConfig holds all configurations for Google provider.
+	GoogleConfig GoogleOptions `json:"googleConfig,omitempty"`
+	// OIDCConfig holds all configurations for OIDC provider
+	// or providers utilize OIDC configurations.
+	OIDCConfig OIDCOptions `json:"oidcConfig,omitempty"`
+	// LoginGovConfig holds all configurations for LoginGov provider.
+	LoginGovConfig LoginGovOptions `json:"loginGovConfig,omitempty"`
+
+	// ProviderID should be a unique identifier for the provider.
+	// This value is required for all providers.
+	ProviderID string `json:"providerID,omitempty"`
+	// ProviderType is the OAuth provider
+	// must be set from the supported providers group,
+	// otherwise 'Google' is set as default
+	ProviderType string `json:"provider,omitempty"`
+	// ProviderName is the providers display name
+	// if set, it will be shown to the users in the login page.
+	ProviderName string `json:"providerDisplayName,omitempty"`
+	// ProviderCAFiles is a list of paths to CA certificates that should be used when connecting to the provider.
+	// If not specified, the default Go trust sources are used instead
+	ProviderCAFiles []string `json:"providerCAFiles,omitempty"`
```

Comment:
I wonder, given this is already in a providers list, whether we need to duplicate `provider` in the name here too?